safe_transaction_service/history/tests/test_models.py

```diff
@@ -1325,6 +1325,69 @@ def test_not_indexed_metadata_contract_addresses(self):
             MultisigTransaction.objects.not_indexed_metadata_contract_addresses()
         )
 
+    def test_with_confirmations_required(self):
+        # This should never be picked
+        SafeStatusFactory(nonce=0, threshold=4)
+
+        multisig_transaction = MultisigTransactionFactory()
+        self.assertIsNone(
+            MultisigTransaction.objects.with_confirmations_required()
+            .first()
+            .confirmations_required
+        )
+
+        # SafeStatus not matching the EthereumTx
+        safe_status = SafeStatusFactory(nonce=1, threshold=8)
+        self.assertIsNone(
+            MultisigTransaction.objects.with_confirmations_required()
+            .first()
+            .confirmations_required
+        )
+
+        safe_status.internal_tx.ethereum_tx = multisig_transaction.ethereum_tx
+        safe_status.internal_tx.save(update_fields=["ethereum_tx"])
+
+        self.assertEqual(
+            MultisigTransaction.objects.with_confirmations_required()
+            .first()
+            .confirmations_required,
+            8,
+        )
+
+        # It will not be picked, as EthereumTx is not matching
+        SafeStatusFactory(nonce=2, threshold=15)
+        self.assertEqual(
+            MultisigTransaction.objects.with_confirmations_required()
+            .first()
+            .confirmations_required,
+            8,
```

Comment:
This part of the test is a little bit confusing for me, could we change `assertEqual` for `assertNotEqual` 15.

Reply:
I don't think that's a good idea. I want to check that `confirmations_required` are now `8` and inserting a new `SafeStatus` didn't change it (maybe to `4` again due to some bad logic implementation). If I take your suggestion I would be missing a test case,